openlibrary/catalog/marc/parse.py

```diff
@@ -442,18 +442,19 @@ def read_author_person(field: MarcFieldBase, tag: str = '100') -> dict | None:
     for subfield, field_name in subfields:
         if subfield in contents:
             author[field_name] = name_from_list(contents[subfield])
```

Comment:
$e subfield needs to be treated specially here because `name_from_list` strips trailing periods which we don't want.

Reply:
@tfmorris thanks for picking this up... I've changed it so that trailing dots are no longer stripped from $e / role, but that does leave "supposed author." with the trailing dot in that one example.

This seems to be the lesser of two evils, and I'm pleasantly surprised that we even have test cases that cover both sentence terminating points and abbreviation points. Abbreviated roles do seem common.

I'm not sure if we can rely on strict ISBD, AACR, or any other punctuation rules to make a correct distinction here. It looks like hybrid / ad-hoc cataloguing punctuation is to be expected in MARC. Unfortunately.

Reply:
@tfmorris @scottbarnes  I believe I've addressed the recent comments and pushed improvements. Is this good  to  merge now?

Reply:
This looks good to me. Thanks for all the work on this @hornc and @tfmorris. I am going to merge this and if there is more to do I vote it is on me to fix it. :)